website/blog/testing.md

```diff
@@ -0,0 +1,604 @@
+# Lower Test Toil With Better Local Testing
+
+[Service Weaver][tutorial] is a programming framework that makes it easy to
+write, test, and deploy distributed applications. In [previous blog
+posts][blog], we discussed how to write and deploy Service Weaver applications.
+In this blog post, we'll focus on testing. Specifically, we'll explore
+
+- how Service Weaver lets you **unit test** system components using regular Go
+  unit tests;
+- how Service Weaver makes it significantly easier to write **integration
+  tests** that would otherwise be slow and brittle; and
+- how Service Weaver implements an advanced form of **randomized testing**
+  called deterministic simulation that can find rare bugs that only emerge in
+  pathological cases.
+
+These three types of testing&mdash;unit testing, integration testing, and
+randomized testing&mdash;all fall on a spectrum trading off test complexity and
+test coverage. All three types of tests are valuable; they complement one
+another.
+
+## 1. Unit Testing
+
+Service Weaver has a [`weavertest` package][unit_testing] that makes it easy to
+**unit test** [components][components]. For example, consider the following
+`Adder` component:
+
+```go
+type Adder interface {
+    Add(context.Context, int, int) (int, error)
+}
+
+type adder struct {
+    weaver.Implements[Adder]
+}
+
+func (*adder) Add(_ context.Context, x, y int) (int, error) {
+    return x + y, nil
+}
+```
+
+We can use the `weavertest` package to unit test the `Adder` component using
+[idiomatic Go unit tests][go_testing]:
+
+```go
+package main
+
+import (
+    "context"
+    "testing"
+
+    "github.com/ServiceWeaver/weaver"
+    "github.com/ServiceWeaver/weaver/weavertest"
+)
+
+func TestAdd(t *testing.T) {
```

Comment:
Can we explain either with words or comments what's the test doing?

Reply:
Done.